sklearn/neighbors/lmnn.py

```diff
@@ -73,6 +73,12 @@ class LargeMarginNearestNeighbor(BaseEstimator, TransformerMixin):
         apart from ``n_neighbors`` - that will be used to select the target
         neighbors.
 
+    weight_push_loss : float, optional (default=0.5)
+        A float in (0, 1], weighting the push loss. This is parameter ``mu``
+        in the journal paper (See references below). Since the pull loss stays
```

Comment:
Here you simply mean that in practice you normalize the objective function such that the push loss has weight 1 (and hence pull loss has weight (1 - mu)/mu? The sentence is a bit confusing: the pull loss itself is not fixed (since it depends on L).

Maybe it is easier for code clarity to simply use the original (1-mu) and mu weights?

Reply:
Hi @bellet , thanks for noting that it's not clear. Yes, that's what I meant, I wrote it like your sentence now. The variable I call ``grad_static`` is the part of the gradient of the pull loss that stays fixed. In the optimization, one only needs to pre-multiply it with L, so the (normalized) weight of the pull loss can be applied once to ``grad_static`` in the beginning.

I would prefer to keep the code as is, since it is a common practice to normalize the objective and if we have your sentence in the docs, it should be straightforward for code readers (?)